src/coreclr/vm/invokeutil.cpp

```diff
@@ -905,10 +907,17 @@ void InvokeUtil::SetValidField(CorElementType fldType,
         MethodTable *pMT = fldTH.AsMethodTable();
         {
             void* pFieldData;
+            fldTH.AsMethodTable()->EnsureActive();
+            obj = fldTH.AsMethodTable()->Allocate();
+            GCPROTECT_BEGIN(obj);
             if (pField->IsStatic())
                 pFieldData = pField->GetCurrentStaticAddress();
             else
-                pFieldData = (*((BYTE**)target)) + pField->GetOffset() + sizeof(Object);
+            {
+                OBJECTREF objRef = ObjectToOBJECTREF(obj);
+                pFieldData = pField->GetAddress(OBJECTREFToObject(objRef));
```

Comment:
```suggestion
            {
               pFieldData = pField->GetInstanceAddress(OBJECTREFToObject(objRef));
```
This should be all that is necessary to fix this.

Reply:
`FieldDesc::GetInstanceAddress` takes care of creating the storage for EnC added field as necessary. We use this method in number of other places. So the fix should be to just call this method here instead of assuming field with an offset.

Reply:
How do I decide whether I pass the target or the object in? The first time, I just used an example of how the function was used, by passing the object in. When I read the comment above the SetFieldValue, it says that the method will set the field on the target object, so it makes sense to have the target be the input.

Reply:
I am not sure what you mean by target. 

We have object instance, FieldDesc, and new field value. We want to:

1. Compute address of the field. It is what `FieldDesc::GetInstanceAddress` does for all types of fields. Notice that `FieldDesc::GetInstanceAddress` implementation checks whether the field is EnC added field first. If it is EnC added field, it will do the special handling of EnC added fields. If it is not EnC added field, it will just do instance + offset like the existing code here.

2. Assign the value to the address computed in the previous step. The existing code does that https://github.com/dotnet/runtime/pull/90446/files#diff-59f60204ba8c710bc8311b010ae4ceb4a856b6ab21f64f7f3e8b6594d432cd52R922-R925.

Reply:
Thank you. I was just referring to Aaron's comment above where he states that we want the address of the target. I assumed that "target" refers to the variable name inside of Get/SetFieldValue called target. 

So, the "object" parameter is the FieldDesc in Get/SetFieldValue?

Reply:
> I assumed that "target" refers to the variable name inside of Get/SetFieldValue called target.

Yep. @jkotas was calling the "instance" and in this case his use of "instance" and the local called `target` are the same thing.